appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -250,14 +250,14 @@ private ComponentInvocation getSavedInvocation() {
      */
     private boolean isApplicationEnabled(String appId) {
         if (appId == null) {
-            // we don't know the application name yet, it is starting.
-            return true;
+            return false;
```

Comment:
This change is wrong - the DOL does still some magic. I think I did something similar when I reintroduced concurrency before JEE10 and I broke some other things by this change. I see also my FIXME in the history:
```
             // we don't know the application name yet, it is starting.
            // FIXME: it would be good to know the application name even in this phase.
```
You moved the comment down, now it even doesn't make sense, because you return true instead original false.
And  javadoc doesn't make sense too - no exception would be thrown, it would return false. Now it returns true.

Reply:
If it all would be written correctly, for null appId it should thrown NPE. We have to get there some day :-)

Reply:
Returning true at the end makes sende - it we have appName but the Application object isn’t created yet, then the app is being deployed and thus it’s enabled. This actually fixes #21087 and it’s a similar fix to how Payara fixed the same issue a while ago.

Reply:
I don’t understand why DOL would call this method with no app name. Then which app is enabled if we return true?

Reply:
> This change is wrong - the DOL does still some magic. I think I did something similar when I reintroduced concurrency before JEE10 and I broke some other things by this change.

Now I see that a test in `application-tests` failed exactly because `appName` is `null` in ContextSetupProviderimpl (in concurrent-impl). Because `invocation.getAppName()` is null. I'll try to find out why that happens.

Reply:
Here's the reason for `invocation.getAppName() == null`:
https://github.com/eclipse-ee4j/glassfish/blob/master/appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java#L243

By default, no information from the current invocation is saved for a new thread if JNDI isn't propagated. getSavedInvocation() method either returns an empty invocation (if JNDI should be cleared), or `null` (by defualt). Both are later interpreted as `appName == null`.

I believe that in both cases, a new invocation should be created and some non-JNDI data should be copied, e.g. app name.

Reply:
OK, I resign for now. I reverted this change. I have an idea how to improve the current concurrency runtime and propagate app context if there's no current context but that would be for another PR.

Reply:
Here's my idea how to propagate app context if there's none set up: https://github.com/eclipse-ee4j/glassfish/pull/24544